marklogic-spark-connector/src/test/java/com/marklogic/spark/writer/classifier/AddConceptsToXmlTest.java

```diff
@@ -63,6 +65,27 @@ void noConceptsAddedWhenNoSemaphoreServerSpecified() {
         doc.assertElementMissing("Expecting the chunk to not include a 'model:concepts' child element", "/root/model:chunks/model:chunk[1]/model:concepts");
     }
 
+    @Test
+    @EnabledIfEnvironmentVariable(named = "SEMAPHORE_API_KEY", matches = ".*")
+    void noHttpsSpecifiedShouldDefaultToHttpAndFail() {
+        final String apiKey = System.getenv("SEMAPHORE_API_KEY");
+        assertNotNull(apiKey);
+
+        final DataFrameWriter<Row> dfw = readDocument("/marklogic-docs/java-client-intro.xml")
+            .write().format(CONNECTOR_IDENTIFIER)
+            .option(Options.CLIENT_URI, makeClientUri())
+            .option(Options.WRITE_SPLITTER_XPATH, "/root/text/text()")
+            .option(Options.WRITE_CLASSIFIER_HOST, "demo.data.progress.cloud")
+            .option(Options.WRITE_CLASSIFIER_PORT, "443")
+            .option(Options.WRITE_CLASSIFIER_ENDPOINT, "/cls/dev/cs1/")
+            .option(Options.WRITE_CLASSIFIER_APIKEY, apiKey)
+            .option(Options.WRITE_CLASSIFIER_TOKEN_ENDPOINT, "token/")
+            .mode(SaveMode.Append);
+
+        ConnectorException exception = assertThrowsConnectorException(dfw::save);
+        assertTrue(exception.getMessage().contains("Error retrieving token"));
```

Comment:
One final nit - a problem with assertTrue/assertFalse is when they fail, like in Jenkins, it's hard to know why because those methods don't show an actual/expected like assertEquals does. So I always include an assertion message of `"Unexpected error: " + exception.getMessage()`.